packages/concerto-core/lib/decoratormanager.js

```diff
@@ -526,14 +646,16 @@ class DecoratorManager {
      */
     static executePropertyCommand(property, command) {
         const { target, decorator, type } = command;
-        if (
-            this.falsyOrEqual(
-                target.property ? target.property : target.properties,
-                [property.name]
-            ) &&
-            this.falsyOrEqual(target.type, [property.$class])
-        ) {
-            this.applyDecorator(property, type, decorator);
+        if(target.properties || target.property || target.type) {
```

Comment:
Does this help with the performance issue that you found?
Are there any non-null, falsy values that need checking here?

Reply:
> Does this help with the performance issue that you found?

There was an edge case where it applied decorators on map elements. This handles that edge case.

>  Are there any non-null, falsy values that need checking here?

Nope, even if there exists that means that it is a malformed command.